pkg/cloudclient/aws/private.go

```diff
@@ -312,6 +312,145 @@ func getMasterNodeSubnets(kclient client.Client) (map[string]string, error) {
 	return subnets, nil
 }
 
+func (c *Client) getPublicSubnets(kclient client.Client) ([]string, error) {
+
+	var publicSubnets []string
+
+	machineList, err := baseutils.GetMasterMachines(kclient)
+
+	if err != nil {
+		log.Error(err, "No master machines found")
+		return nil, err
+	}
+
+	// Get the first master machine in the list
+	masterMachine := machineList.Items[0]
+
+	// Get the AWS ProviderSpec from the machine
+	machineProviderSpec, err := getAWSDecodedProviderSpec(masterMachine)
+	if err != nil {
+		log.Error(err, "Cannot decode AWS Provider Spec")
+		return nil, err
+	}
+
+	// Get the subnet the master machine is in
+	initalSubnet := machineProviderSpec.Subnet.ID
+
+	// Look up the subnet metadata
+	describeSubnetOutput, err := c.ec2Client.DescribeSubnets(&ec2.DescribeSubnetsInput{SubnetIds: []*string{initalSubnet}})
+
+	// Extract the VPC ID from the subnet metadata
+	targetVPC := describeSubnetOutput.Subnets[0].VpcId
+
+	// List all subnets in the VPC
+	allSubnets, err := c.getAllSubnetsInVPC(*targetVPC)
+
+	// List all route tables associated with the VPC
+	routeTables, err := c.getAllRouteTablesInVPC(*targetVPC)
+
+	for _, subnet := range allSubnets {
+		isPublic, err := isSubnetPublic(routeTables, *subnet.SubnetId)
+
+		if err != nil {
+			log.Error(err, "Error while determining if subnet is public")
+			return nil, err
+		}
+		if isPublic {
+			publicSubnets = append(publicSubnets, *subnet.SubnetId)
+		}
+	}
+
+	return publicSubnets, nil
+}
+
+func (c *Client) getAllSubnetsInVPC(vpcID string) ([]*ec2.Subnet, error) {
+
+	var subnetIDs []*ec2.Subnet
+	token := aws.String("initString")
+
+	for token != nil {
+		describeSubnetOutput, err := c.ec2Client.DescribeSubnets(&ec2.DescribeSubnetsInput{Filters: []*ec2.Filter{{Name: aws.String("vpc-id"), Values: []*string{aws.String(vpcID)}}}})
```

Comment:
```suggestion
		describeSubnetOutput, err := c.ec2Client.DescribeSubnets(
			&ec2.DescribeSubnetsInput{
				Filters: []*ec2.Filter{
					{
						Name:   aws.String("vpc-id"),
						Values: []*string{aws.String(vpcID)},
					},
				},
			})
```

This is a pretty long line with a lot going on. Breaking it up, perhaps like this, will aid readability.